### ISSUE TITLE
header points to request in response block / wrong resolve strategy for @DelegatesTo

### ISSUE THREAD

**Issue by KrzysztofKowalczyk** [cjstehno/ersatz#102]
Ersatz is using default delegation strategy for closures, which delegates first to owner of the closure, not to the delegate which in my opinion would be working better.

To add header to response one have to use delegate:

```groovy
server.expectations {
  get("/foo") {
    header('Accept', 'application/json')

    responder {
      delegate.header("Bad", "code") // without delegate that would resolve to Request.header
      content('{"hello":"world"}', 'application/json')
    }
  }
}
```

If ersatz would be marked to do delegate first, one could write:

```groovy
server.expectations {
  get("/foo") {
    header('Accept', 'application/json')

    responder {
      header("Bad", "code") // delegate first resolves to Response.header
      content('{"hello":"world"}', 'application/json')
    }
  }
}
```

**Comment by cjstehno** [comment]
Odd that hasn't popped up earlier. I'll take a look. Thanks.

**Comment by cjstehno** [comment]
Definitely an issue. I'll fix this in the upcoming 1.8 release. Good catch. I tend to forget about the delegate strategy.

**Comment by KrzysztofKowalczyk** [comment]
Thanks for fast reaction

**Comment by cjstehno** [comment]
The fix for this is in release 1.8.1, which was published today.
